stock_packaging_calculator/models/product.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import models
+
+
+class Product(models.Model):
+    _inherit = "product.product"
+
+    def product_qty_by_packaging(self, prod_qty, min_unit=None):
+        """Calculate quantity by packaging.
+
+        :prod_qty: total qty to satisfy.
+        :min_unit: minimal unit of measure as a tuple (qty, name).
+                   Default: to UoM unit.
+        :returns: list of tuple in the form [(qty_per_package, package_name)]
+        """
+        packagings = [(x.qty, x.name) for x in self.packaging_ids]
+        if min_unit is None:
+            # You can pass `False` to skip it.
+            single_unit = self.env.ref("uom.product_uom_unit")
+            min_unit = (1.0, single_unit.name)
+        if min_unit:
+            packagings.append(min_unit)
```

Comment:
No need to check that `min_unit` isn't already in `packagings`? Or use a `set`?

Reply:
It was misleading, we are mixing packaging with uom here OK, so `min_unit` can't be in `packaging`.

Reply:
eh, the point is that very likely you don't have a packaging for the single unit, that's why I'm just using names and allow you to provide different unit if really needed.